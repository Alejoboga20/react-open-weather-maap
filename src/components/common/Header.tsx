import '../../styles/components/Header.css';

export interface HeaderProps {
  title: string;
  subtitle?: string;
}

export const Header = ({ title, subtitle = '' }: HeaderProps) => {
  return (
    <div className='header_container'>
      <h1 className='title'>{title}</h1>
      <h2 className='subtitle'>{subtitle}</h2>
      <hr />
    </div>
  );
};
