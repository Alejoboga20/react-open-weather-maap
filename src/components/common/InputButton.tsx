import { Coordinates } from '../../interfaces/Locations';
import '../../styles/components/InputButton.css';

export interface InputButtonProps {
  label: string;
  value: string;
  location: Coordinates;
  onClick: (event: any) => void;
  disabled: boolean;
}

export const InputButton = ({
  label,
  value,
  onClick,
  disabled,
}: InputButtonProps) => (
  <button
    value={value}
    onClick={onClick}
    disabled={disabled}
    className='inputButton'
  >
    {label}
  </button>
);
