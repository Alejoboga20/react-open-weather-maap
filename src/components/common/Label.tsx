import '../../styles/components/Label.css';

export interface LabelProps {
  labelContent: string;
}

export const Label = ({ labelContent }: LabelProps) => {
  return (
    <div className='label_container'>
      <label className='main_content'>{labelContent}</label>
    </div>
  );
};
