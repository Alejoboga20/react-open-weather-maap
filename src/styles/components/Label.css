.label_container {
  font-size: x-large;
  padding: 15px;
  margin: 10px 0;
}
