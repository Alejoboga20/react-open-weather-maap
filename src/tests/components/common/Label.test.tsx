import { shallow } from 'enzyme';
import { Label, LabelProps } from '../../../components/common/Label';

const labelProps: LabelProps = {
  labelContent: 'Label Tests',
};

describe('Label Tests', () => {
  const wrapper = shallow(<Label {...labelProps} />);

  test('should Match Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
