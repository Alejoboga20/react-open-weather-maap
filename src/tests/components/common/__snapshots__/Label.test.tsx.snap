// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Tests should Match Snapshot 1`] = `
<div
  className="label_container"
>
  <label
    className="main_content"
  >
    Label Tests
  </label>
</div>
`;
